Include/cpython/pyerrors.h

```diff
@@ -7,7 +7,7 @@
 /* PyException_HEAD defines the initial segment of every exception class. */
 #define PyException_HEAD PyObject_HEAD PyObject *dict;\
              PyObject *args; PyObject *note; PyObject *traceback;\
-             PyObject *context; PyObject *cause;\
+             PyObject *context; PyObject* newargs; PyObject *cause;\
```

Comment:
You can't do this. You're creating references to all the args, which may cause memory leaks. This may well be the reason that the programmer is not keeping references to the args but deriving something from them.